crates/iota-graphql-rpc/src/mutation.rs

```diff
@@ -90,28 +116,11 @@ impl Mutation {
             .map_err(|e| Error::Internal(format!("Unable to execute transaction: {e}")))
             .extend()?;
 
-        let native: NativeTransactionEffects = bcs::from_bytes(&result.raw_effects)
-            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction effects: {e}")))
-            .extend()?;
-        let tx_data: SenderSignedData = bcs::from_bytes(&result.raw_transaction)
-            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction data: {e}")))
-            .extend()?;
+        let tx_digest = result.digest;
+        let digest_bytes = tx_digest.inner().to_vec();
 
-        let events = result
-            .events
-            .ok_or_else(|| {
-                Error::Internal("No events are returned from transaction execution".to_string())
-            })?
-            .data
-            .into_iter()
-            .map(|e| NativeEvent {
-                package_id: e.package_id,
-                transaction_module: e.transaction_module,
-                sender: e.sender,
-                type_: e.type_,
-                contents: e.bcs.into_bytes(),
-            })
-            .collect();
+        let db: &Db = ctx.data_unchecked();
+        let optimistic_tx = query_optimistic_transaction_by_digest(db, digest_bytes).await?;
```

Comment:
This seems like an indirection. We already construct the `optimistic_tx` before indexing, so there is room for optimization here. Would possibly require adding optional fields in `IotaTransctionBlockResponse` and the respective `Options`. I suggest we revisit in a new issue.

Reply:
That's true. Created issue for this: https://github.com/iotaledger/iota/issues/8775